### DISCUSSION TITLE
So Chrome is written in C++...

### DISCUSSION THREAD

**Discussion by Tennessene** [emscripten-core/emscripten#20024] [General]
Maybe you see where I'm going here. Has it ever occurred to you that -- theoretically, you could make a web browser inside of a web browser by compiling Chrome's source code? I don't know, but for some reason, I have an urge to try it out. It would be a crazy little project, but everyone likes to have fun with stuff. Chrome inside of Chrome sounds cool, but it also sounds like it wouldn't work. 

Would this even be possible? How would networking even work? Let me know what you think

**Comment by juj** [comment]
There is much more to a computing platform than just the programming language. The programming language is a rather small part.

Like you already noticed, networking interfaces are one thing. Beyond that, multiprocessing, blocking execution model, synchronous thread signaling and memory virtualization and protection are other aspects that come to mind that do not exist in WebAssembly for parity.

After that, there are a number of different specific API areas that have limitations or differences. Though on a large number of those areas, one could probably refactor the codebase to make ends meet.

**Reply by Tennessene** [reply]
Yeah, I understand now. I just didn't know if Enscripten's mini os had the capability to do that. Thank you for the helpful response
